spark-operator-api/src/main/java/org/apache/spark/k8s/operator/utils/ModelUtils.java

```diff
@@ -0,0 +1,115 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.k8s.operator.utils;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.ContainerStatus;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.OwnerReference;
+import io.fabric8.kubernetes.api.model.OwnerReferenceBuilder;
+import io.fabric8.kubernetes.api.model.Pod;
+import io.fabric8.kubernetes.api.model.PodBuilder;
+import io.fabric8.kubernetes.api.model.PodTemplateSpec;
+import org.apache.commons.lang3.StringUtils;
+
+import org.apache.spark.k8s.operator.spec.ApplicationSpec;
+
+public class ModelUtils {
+  public static final String DRIVER_SPARK_CONTAINER_PROP_KEY =
+      "spark.kubernetes.driver.podTemplateContainerName";
+  public static final String DRIVER_SPARK_TEMPLATE_FILE_PROP_KEY =
+      "spark.kubernetes.driver.podTemplateFile";
+  public static final String EXECUTOR_SPARK_TEMPLATE_FILE_PROP_KEY =
+      "spark.kubernetes.executor.podTemplateFile";
+  public static final ObjectMapper objectMapper = new ObjectMapper();
+
+  public static Pod getPodFromTemplateSpec(PodTemplateSpec podTemplateSpec) {
+    if (podTemplateSpec != null) {
+      return new PodBuilder()
+          .withMetadata(podTemplateSpec.getMetadata())
+          .withSpec(podTemplateSpec.getSpec())
+          .withAdditionalProperties(podTemplateSpec.getAdditionalProperties())
+          .build();
+    } else {
+      return new PodBuilder().withNewMetadata().endMetadata().withNewSpec().endSpec().build();
+    }
+  }
+
+  /**
+   * Find the Spark main container(s) in driver pod. If `spark.kubernetes.driver
+   * .podTemplateContainerName` is not set, all containers are considered as main container from
+   * health monitoring perspective
+   */
+  public static List<ContainerStatus> findDriverMainContainerStatus(
+      final ApplicationSpec appSpec, final List<ContainerStatus> containerStatusList) {
+    if (appSpec == null
+        || appSpec.getSparkConf() == null
+        || !appSpec.getSparkConf().containsKey(DRIVER_SPARK_CONTAINER_PROP_KEY)) {
```

Comment:
Although I know that this came from Oracle Java guideline, `Apache Spark` community prefers to start with words like the following. Shall we change the style?
```java
if (appSpec == null ||
    appSpec.getSparkConf() == null ||
    !appSpec.getSparkConf().containsKey(DRIVER_SPARK_CONTAINER_PROP_KEY)) {
```

Reply:
This is also introduced & enforced by spotless when following [Google Java style guide](https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break), which behaves different from [scala style guide](https://docs.scala-lang.org/style/indentation.html#line-wrapping) , or other languages like C++ when wrapping lines. Google Java style guide is widely used by other java projects like Lucene / Flink. 

Can we consider this as an exception from style align due to language difference ??

Reply:
Got it. If you think so and we can maintain this style strongly in this repo, I'm also +1 for your idea.